### PR TITLE
Feature/submitting time

### PR DESCRIPTION
**Pull request by 7ep**
New capability to submit a time period, in progress

no UI yet

Co-authored-by: Matthew Taylor <matthew.taylor@coveros.com>
Co-authored-by: Richard Mills <rich.mills@coveros.com>
Co-authored-by: Mitch Hardee <mitchell.hardee@coveros.com>

### CODE REVIEW

File: src/main/kotlin/coverosR3z/persistence/utility/DatabaseDiskPersistence.kt [hunk 1]
```diff
@@ -139,7 +139,7 @@ class DatabaseDiskPersistence(private val dbDirectory : String? = null) {
             val employees = readAndDeserialize(dbDirectory, Employee.directoryName) { Employee.Deserializer().deserialize(it) }
             val timeEntries = readAndDeserialize(dbDirectory, TimeEntry.directoryName) { TimeEntry.Deserializer(employees, projects).deserialize(it) }
 
-            return PureMemoryDatabase(employees, users, projects, timeEntries, sessions, dbDirectory)
+            return PureMemoryDatabase(employees, users, projects, timeEntries, sessions, ChangeTrackingSet(), dbDirectory)
```
Comment (7ep) [inline src/main/kotlin/coverosR3z/persistence/utility/DatabaseDiskPersistence.kt:142]: This needs to be made more real

File: src/main/kotlin/coverosR3z/persistence/utility/PureMemoryDatabase.kt [hunk 2]
```diff
@@ -23,12 +20,14 @@ import coverosR3z.timerecording.types.TimeEntry
...
-                         protected val sessions: ChangeTrackingSet<Session> = ChangeTrackingSet(),
-                         protected val dbDirectory : String? = null
+open class PureMemoryDatabase(
+    protected val employees: ChangeTrackingSet<Employee> = ChangeTrackingSet(),
+    protected val users: ChangeTrackingSet<User> = ChangeTrackingSet(),
+    protected val projects: ChangeTrackingSet<Project> = ChangeTrackingSet(),
+    protected val timeEntries: ChangeTrackingSet<TimeEntry> = ChangeTrackingSet(),
+    protected val sessions: ChangeTrackingSet<Session> = ChangeTrackingSet(),
+    protected val submittedPeriods: ChangeTrackingSet<SubmittedPeriod> = ChangeTrackingSet(),
+    protected val dbDirectory : String? = null
```
Comment (7ep) [inline src/main/kotlin/coverosR3z/persistence/utility/PureMemoryDatabase.kt:30]: Starting to get a bit heavy with all these params

File: src/main/kotlin/coverosR3z/persistence/utility/PureMemoryDatabase.kt [hunk 3]
```diff
@@ -41,6 +40,7 @@ open class PureMemoryDatabase(protected val employees: ChangeTrackingSet<Employe
             projects = this.projects.toList().toChangeTrackingSet(),
             timeEntries = this.timeEntries.toList().toChangeTrackingSet(),
             sessions = this.sessions.toList().toChangeTrackingSet(),
+            submittedPeriods = this.submittedPeriods.toList().toChangeTrackingSet(),
```
Comment (7ep) [inline src/main/kotlin/coverosR3z/persistence/utility/PureMemoryDatabase.kt:43]: is it possible to remove the boilerplate?

File: src/main/kotlin/coverosR3z/timerecording/api/SubmitTimeAPI.kt [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    // internal handlePOST() to do the work.
+    fun handlePOST() : PreparedResponseData {
+        val data = sd.ahd.data
+        val tru = sd.tru // time recording utilities
+        val startDate = Date.make(data.mapping[SubmitTimeAPI.Elements.START_DATE.getElemName()])
+        val endDate = Date.make(data.mapping[SubmitTimeAPI.Elements.END_DATE.getElemName()])
+        val employee = tru.findEmployeeById(checkNotNull(sd.ahd.user.employeeId){ employeeIdNotNullMsg })
+
+        // UNFINISHED: Have not implemented time periods yet.
```
Comment (7ep) [inline src/main/kotlin/coverosR3z/timerecording/api/SubmitTimeAPI.kt:56]: this will be next to handle

**Review by coveros-matt-taylor — APPROVED** [review]
Lookin submittable!

**Pull request merged**